sy-server/migrations/2020-07-23-031356_users/up.sql

```diff
@@ -1,5 +1,49 @@
 CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
 
```

Comment:
Main thing to add here is that you should create a separate migration by running `diesel migration generate <name>` and not modify an existing migration.

Reply:
Thanks for the clarification.. one user max assigned to task is a good idea. I'll fix the schema.

Reply:
Scenario: New user, creates a task, doesn't assign it to anyone. User has not joined/created a group. Since task's group_id can't be null, what now? Should new users be automatically assigned to a group?

Reply:
I am also confused on how to select migrations:
agasfer@Lukich:~/Projects/grpc-rust-dart-demo/sy-server$ diesel migration list
Migrations:
  [X] 2020-07-23-031356_users
  [ ] 2020-09-14-235351_tasks_messages

How do I select the second one?

Reply:
diesel migration run.  to revert you run diesel migration revert

Reply:
> Scenario: New user, creates a task, doesn't assign it to anyone. User has not joined/created a group. Since task's group_id can't be null, what now? Should new users be automatically assigned to a group?

Both good questions.  I can definitely see cases when users do not belong to a group.  If you have group to be never null, you would have to create a default group for the user.  I'm not sure.  The scenarios I was thinking were: User creates account, create default group.  User gets invited, they get to be part of that group.  This way the user is always part of a group therefore task should never not belong to a group.

What do you think?

Reply:
I thought about a default group, but what would be the benefit? Just to not have a null group_id field in the tasks table? What are the downsides to having a nullable group_id field since we already have nullable assigned_user_id field? The owner_id field is not nullable so we shouldn't have any lost tasks.